lms/djangoapps/discussion_api/permissions.py

```diff
@@ -18,6 +20,34 @@ def _is_author_or_privileged(cc_content, context):
     return context["is_requester_privileged"] or _is_author(cc_content, context)
 
 
+NON_UPDATABLE_THREAD_FIELDS = {"course_id"}
+NON_UPDATABLE_COMMENT_FIELDS = {"thread_id", "parent_id"}
+
+
+def get_initializable_thread_fields(context):
+    """
+    Return the set of fields that the requester can initialize for a thread
+    """
+    ret = get_editable_fields(
+        Thread(user_id=context["cc_requester"]["id"], type="thread"),
```

Comment:
It is a bit roundabout, but I like it in that it neatly expresses the idea that you can initialize any field that you can edit on a thread that you author. Perhaps a comment to that effect would make it clearer.